project_event/models/project_task.py

```diff
@@ -286,7 +286,26 @@ def write(self, vals):
                 task.write({'responsible_id': self.responsible_id.id,
                            'partner_id': self.partner_id.id})
         elif self.activity_task_type == 'task':
-            return super(Task, self).write(vals)
+            self.ensure_one()
```

Comment:
If you want to ensure we have a singleton, this self.ensure_one() should go at the beginning of the method, not here. But, what would be better is to create a new method, because the write method is already overwhelmed. it does too many things. Better to put your feature in a new method and use it when writing

Reply:
@lgharib This method refactor completely so you need to test and change your code accordingly